docs/Documentation/Development/Build_Tools/git.md

```diff
@@ -4,20 +4,196 @@ title: Git
+
+??? note "Fork" 
+    A fork is a personal copy of another user's repository that lives on your account. Forks allow you to freely make changes to a project without affecting the original. Forks remain attached to the original, allowing you to submit a pull request to the original's author to update with your changes. You can also keep your fork up to date by pulling in updates from the original.
+
+??? note "Pull"
+    Pull refers to when you are fetching in changes and merging them. For instance, if someone has edited the remote file you're both working on, you'll want to pull in those changes to your local copy so that it's up to date.
+    
+??? note "Pull Request"
+    Pull requests are proposed changes to a repository submitted by a user and accepted or rejected by a repository's collaborators. Like issues, pull requests each have their own discussion forum. 
+
+??? note "Push"
+    Pushing refers to sending your committed changes to a remote repository, such as a repository hosted on GitHub. For instance, if you change something locally, you'd want to then push those changes so that others may access them.
+
+??? note "Branch"
+    A branch is a new/separate version of the repository. Use branches when you want to work on a new feature, but don't want to mess-up the main branch while testing your ideas. 
+## Tool Use
+
+??? note "Clone an existing repo"
+    For example, you could create a local working copy of the "test_repo" repo (puts it in a folder in your current directory):
+    ```
+    cd /some/project/dir
+    git clone <git@github.nrel.gov:username/test_repo.git>
+    ```
+    Now, make changes to whatever you need to work on.
+    Recommendation: commit your changes often, e.g., whenever you have a workable chunk of work completed.
+
+??? note "See what files you've changed"
+    `git status`
+
+??? note "Push your changes to the repo"
+    ```
+    git add <filename(s)-you-changed>
+    git commit -m "A comment about the changes you just made."
+    git push
+    ```
+??? note "Get remote changes from the repo"
+    If you collaborate with others in this repo, you'll want to pull their changes into your copy of the repo. You may want to do this first-thing when you sit down to work on something to minimize the number of merges you'll need to handle:
+    `git pull`
+
+??? note "Create a new local git code repo"
+    ```
+    mkdir my.projectname
+    cd my.projectname
+    git init
+    touch README.txt
+    git add README.txt
+    git commit -m 'first commit'
+    # Push the repo to Github
+    git remote add origin git@hpc/my.projectname.git
+    git push origin main
+    ```
+
+??? note "Revert to a previous commit"
+    Find the commit you want to reset to:  
+    `git log`  
+    Once you have the hash:  
+    `git reset --hard <hash>`  
+    And to push onto the remote:  
+    `git push -f <remote> <branch>`  
```

Comment:
Depending on the intent of this blurb, I would suggest either removing this example or using a different approach.  I would be hesitant to include a `git reset --hard` in a general guidance doc, as that can cause one to lose data (history), and should be used very cautiously.  And I definitely would not include an example with force pushing, as that should almost never be done unless someone really knows what they are doing (it overwrites history on a remote).

If the idea here is to undo the changes from a particular commit, there is `git revert`, which doesn't change history and is safer than `reset --hard`.  Or another approach is to create a new branch as a way of going back to a previous point in history, e.g., `git branch <branchname> <commit>`.

But once the changes in question get into a public version of the repo, history should not be changed with a force push.  So undoing public changes would mean either using git revert or manually creating new commits to undo the changes.

Reply:
That's a good point. I wasn't the original author of this section, so I'm not sure of the intent, but changed to include `git revert` instead. I agree that's a better method.